addon/components/bourbon-text-field.js

```diff
@@ -29,28 +37,43 @@ export default TextField.extend({
   }),
 
   focusedElementObserver: observer('autofocus', function() {
-    this.set('isFocused', this.get('autofocus'))
+    this.set('isFocused', this.get('autofocus'));
+  }),
+
+  valueObserver: observer('value', function() {
+    if (this.get('value') && !this.get('noLabel')) {
+      this.set('isNotEmpty', true);
+    } else {
+      this.set('isNotEmpty', false);
+    }
   }),
 
-  didInsertElement()  {
+  didInsertElement() {
     if (this.get('autofocus')) {
-      this.$().focus()
+      this.$().focus();
       this.set('isFocused', true);
     }
   },
 
   focusIn() {
-    this.set('isFocused', true)
+    this.set('isFocused', true);
     if (this.get('actionOnFocusIn')) {
       this.get('actionOnFocusIn')();
     }
   },
 
+  input(e) {
+    let el = $(e.currentTarget);
+    let textInput = el.find('.BourbonTextField-input').val();
+
+    this.set('value', textInput);
+  },
+
   focusOut() {
-    this.set('isFocused', false)
+    this.set('isFocused', false);
 
     if (this.get('onFocusOutOrEnter')) {
-      this.get('onFocusOutOrEnter')(this.get('value'))
+      this.get('onFocusOutOrEnter')(this.get('value'));
```

Comment:
It looks like you're calling `this.get('onFocusOutOrEnter')` as a function, but that property's default value is an empty string. How does this actually work?

Reply:
the default property should be null on this.. thanks for catching that.  :)